pymc4/distributions/continuous.py

```diff
@@ -1165,6 +1169,103 @@ def upper_limit(self):
         return self.conditions["high"]
 
 
+class Flat(ContinuousDistribution):
+    r"""A uniform distribution with support :math:`(-\inf, \inf)`.
+    Used as a uninformative log-likelihood that returns
+    zeros regardless of the passed values.
+    """
+
+    def __init__(self, name, **kwargs):
+        super().__init__(name, **kwargs)
+
+    @staticmethod
+    def _init_distribution(conditions):
+        return tfd.Uniform(low=-np.inf, high=np.inf)
+
+    def log_prob(self, value):
+        # convert the value to tensor
+        value = tf.convert_to_tensor(value)
+        # if not tf.rank(value):
+        #     value = tf.convert_to_tensor([value])
+        # value_shape = value.shape
+        # # if the rank of the `value` is less than distribution's `event shape` raise an error
+        # if tf.rank(value) < tf.rank(self._distribution.event_shape):
+        #     raise ValueError("Rank of input tensor less than distribution's event shape")
+        # # if the rightmost axis of `value` doesn't match the distribution's `event_shape`, raise an error
+        # if (
+        #     len(self._distribution.event_shape)
```

Comment:
> 2. There is no tfp equivalent, so we must test that the returned `log_prob` values are correct. That sounds simple because the log_prob should always be zero. The difficulty here is shape handling. If the `value` passed to `log_prob` has a rank that is lower than the distribution's event shape, we should raise an error, if the value has a shape that doesn't broadcast with the distribution's batch+event shape, we should raise an error. Actually, I think it's even more stringent than that, the rightmost axis of the passed values should exactly match the distribution's event shape, not just broadcast with it, or an error should be raised. Furthermore, the log prob should sum reduce over the event shape axis, in order to only get back a batch of zeroes.

As the distribution is univariate the event_shape and batch_shape will always be ``()`` and so I think we don't need to worry about it, right?

Reply:
The defaults will be `()`, but we can stack multiple independent variable that follow this distribution with something similar to `tfd.Sample`, and that will lead to an event shape that could be a tuple of any length.
Furthermore, in #167, we are also aiming to provide a similar mechanism as `tfd.Sample` but instead of stacking events, we will stack batches. So in the end, any distribution can have any number of event and batch axes. I just realized that I wrote the singular axis in my previous comment, but I meant it to be the plural, axes.